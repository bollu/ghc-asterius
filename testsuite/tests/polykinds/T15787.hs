{-# Language RankNTypes #-}
{-# Language TypeApplications #-}
{-# Language DataKinds        #-}
{-# Language PolyKinds        #-}
{-# Language GADTs            #-}
{-# Language TypeFamilies     #-}

import Data.Kind

class RÃ­ki (ob :: Type) where
 type Hom :: ob -> ob -> Type

data
  Kl_kind :: forall ob . (ob -> ob) -> ob -> Type where
  Kl      :: k -> Kl_kind (m :: ob -> ob) k

type family
  UnKl (kl :: Kl_kind m k) = (res :: k) where
  UnKl (Kl a) = a
