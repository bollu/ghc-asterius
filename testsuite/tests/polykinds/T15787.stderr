
T15787.hs:15:43: error:
    • Expected kind ‘ob’, but ‘k’ has kind ‘*’
    • In the second argument of ‘Kl_kind’, namely ‘k’
      In the type ‘Kl_kind (m :: ob -> ob) k’
      In the definition of data constructor ‘Kl’
